meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/util/ZkMetaClientUtil.java

```diff
@@ -262,6 +262,14 @@ public static MetaClientInterface.ConnectState translateKeeperStateToMetaClientC
     }
   }
 
+  public static MetaClientInterface.Stat convertZkStatToStat(
+      org.apache.zookeeper.data.Stat zkStat) {
+    return new MetaClientInterface.Stat(
+        convertZkEntryModeToMetaClientEntryMode(zkStat.getEphemeralOwner()), zkStat.getVersion(),
```

Comment:
I am slightly confused on why this is needed. I think we already have a method` exists(key)` that converts the stat from a node at path key. Is there a reason we can't use that above?

Reply:
exists(key) fetch stat from ZK and translate to MetaClientInterface.Stat.
The new API fetch data and stat in one go so no change happens in between. 

Good point on code duplication, I will change  `exists(key)` to use this Util.